src/Compilers/CSharp/Portable/Lowering/ExtensionMethodBodyRewriter.cs

```diff
@@ -96,6 +94,9 @@ public override ParameterSymbol VisitParameterSymbol(ParameterSymbol symbol)
         {
             var rewritten = new RewrittenLambdaOrLocalFunctionSymbol(node.Symbol, _rewrittenContainingMethod);
 
+            // BoundMethodDefIndex in instrumentation will refer to the lambda method symbol, so we need to map it.
+            _symbolMap = _symbolMap.Add(node.Symbol, rewritten);
```

Comment:
>_symbolMap [](http://example.com/codeflow?start=12&length=10)

Should the map be restored at the end?

Reply:
I didn't see harm in leaving it in scope